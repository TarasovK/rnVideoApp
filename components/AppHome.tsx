import React from 'react'
import { View, StyleSheet, SafeAreaView, Text, Dimensions } from 'react-native'
import { Toggle } from './Toggle'
import { ToggleType } from '../api/ToggleType'
import { Footer } from './Footer'
import { Recomendations } from './Recomendations'
import  { AnimatedComments }  from './AnimatedComments'

const windowHeight = Dimensions.get('window').height;

interface StateHome {
    activeScreen: ToggleType,
    showComments: boolean,
}

export class AppHome extends React.Component<{}, StateHome> {
    constructor(props: {}) {
        super(props)
        this.state = {
            activeScreen: ToggleType.recomedations,
            showComments: false
        }
    }

    renderMainScreen = () => {
        return this.state.activeScreen === ToggleType.recomedations ? <Recomendations onCommentsPress={this.toggleComments}/> : this.renderSubscriptions();
    }

    renderSubscriptions = () => {
        return (
            <Text style={styles.subscriptionsRoot}> 
                    This is a subscriptions view
                </Text>
        )
    }    

    onScreenToggle = (selectedItem: ToggleType)  => {
        if(this.state.activeScreen !== selectedItem) {
            this.setState({activeScreen: selectedItem})
        }        
    }    
    
    toggleComments = () => {
        this.setState({
            showComments: !this.state.showComments
        })
    }

    renderComments = () => {
        if(this.state.showComments) {
            return <AnimatedComments defaultPosition={windowHeight} targetPosition={0} animationCallback={this.toggleComments}/>           
        }        
    }  
    
    render() {
        return(
            <SafeAreaView style={styles.root}>                   
                <View style={styles.playerView}>
                    {this.renderMainScreen()}  
                </View>
                <View style={styles.headerView}>
                    <Toggle 
                        onPress={this.onScreenToggle}
                        defaultScreen={this.state.activeScreen}/>    
                </View>
                <View style={styles.footerView}>
                    <Footer/>                              
                </View>      
                {this.renderComments()}                                      
            </SafeAreaView>
        )
    }
}

var styles = StyleSheet.create({
    root: {
        flex: 1,
        alignItems: "center",
        backgroundColor: '#000'
    },         
    subscriptionsRoot: {
        flex: 1,
        justifyContent: "center",
        color: '#FFF',
        textAlignVertical: 'center'
    },
    playerView: {
        alignItems: "center",
        justifyContent: "center",
        position: "absolute",
        top: 0,
        left: 0,
        bottom: 0, 
        right: 0
    },
    headerView: {
        position: "absolute",
        top: 0,
        left: 0,
        right: 0
    },
    footerView: {
        position: "absolute",
        bottom: 0,
        left: 0,
        right: 0
    },
    closeCommentsView: {
        width: '100%',
        height: '20%',
        backgroundColor: 'transparent',
    },
    commentsRoot: {        
        height: '100%',
        width: '100%',
        borderColor: '#0F0',
    },
    commentsSection: {
        flex: 1,
        width: '100%',
        height: '80%',
    },
});