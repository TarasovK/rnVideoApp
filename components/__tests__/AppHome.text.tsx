import React from 'react';
import { AppHome }  from '../AppHome';
import { render } from '@testing-library/react-native';

import renderer from 'react-test-renderer';

jest.mock('react-native-video', () => 'Video')

it('renders correctly', () => {
  const tree = renderer.create(<AppHome />);
  expect(tree).toMatchSnapshot()
});


it('contains navigation elements', async () => {
    const {findAllByText} = render(<AppHome />);

    // const itemAppHome = await findAllByText(/AppHome/)
    // expect(itemAppHome).toBeTruthy()

    const itemRec = await findAllByText(/Recommendations/)
    expect(itemRec).toBeTruthy()

    const itemSub = await findAllByText(/My Subscriptions/)
    expect(itemSub).toBeTruthy()

    const itemHome = await findAllByText(/Home/)
    expect(itemHome).toBeTruthy()

    const itemSearch = await findAllByText(/Search/)
    expect(itemSearch).toBeTruthy()

    const itemMsg = await findAllByText(/Messages/)
    expect(itemMsg).toBeTruthy()

    const itemProfile = await findAllByText(/Profile/)
    expect(itemProfile).toBeTruthy()
});