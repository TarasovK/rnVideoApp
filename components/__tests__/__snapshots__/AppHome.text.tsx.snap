// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#0FF",
          "borderWidth": 0,
          "flex": 1,
          "justifyContent": "center",
          "zIndex": 0,
        }
      }
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "id": "0",
              "poster": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/images/ForBiggerBlazes.jpg",
              "uri": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4",
            },
            Object {
              "id": "1",
              "poster": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/images/BigBuckBunny.jpg",
              "uri": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
            },
            Object {
              "id": "2",
              "poster": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/images/ForBiggerMeltdowns.jpg",
              "uri": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerMeltdowns.mp4",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onViewableItemsChanged={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfig={
          Object {
            "itemVisiblePercentThreshold": 50,
          }
        }
        viewabilityConfigCallbackPairs={
          Array [
            Object {
              "onViewableItemsChanged": [Function],
              "viewabilityConfig": Object {
                "itemVisiblePercentThreshold": 50,
              },
            },
          ]
        }
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <Video
                paused={true}
                repeat={true}
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                style={
                  Object {
                    "height": 320,
                    "position": "absolute",
                    "right": 0,
                    "top": 507,
                    "width": 80,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/author.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Author
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/like.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      1.8m
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/comments.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Comments
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/share.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Share
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <Video
                paused={true}
                repeat={true}
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                style={
                  Object {
                    "height": 320,
                    "position": "absolute",
                    "right": 0,
                    "top": 507,
                    "width": 80,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/author.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Author
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/like.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      1.8m
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/comments.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Comments
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/share.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Share
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <Video
                paused={true}
                repeat={true}
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerMeltdowns.mp4",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                style={
                  Object {
                    "height": 320,
                    "position": "absolute",
                    "right": 0,
                    "top": 507,
                    "width": 80,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/author.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Author
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/like.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      1.8m
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/comments.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Comments
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "opacity": 1,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../components/icons/share.png",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "contain",
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "#FFF",
                          "flex": 1,
                          "fontSize": 12,
                          "textAlign": "center",
                        }
                      }
                    >
                      Share
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginTop": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "textAlign": "right",
                },
                Object {
                  "color": "#AAA",
                  "fontSize": 14,
                  "margin": 6,
                },
              ]
            }
          >
            My Subscriptions
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "#FFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "margin": 6,
                },
              ]
            }
          >
            Recommendations
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": 70,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../components/icons/home.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#FFF",
              "flex": 1,
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../components/icons/search.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#FFF",
              "flex": 1,
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          Search
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../components/icons/plus.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#FFF",
              "flex": 1,
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../components/icons/messages.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#FFF",
              "flex": 1,
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          Messages
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../components/icons/profile.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "resizeMode": "contain",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#FFF",
              "flex": 1,
              "fontSize": 12,
              "textAlign": "center",
            }
          }
        >
          Profile
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
